docs/proposals/dedicated-nodes.md

```diff
@@ -0,0 +1,298 @@
+To put it another way, with taints and tolerations, only pods that use a set of special nodes
+need to know about those special nodes; with the HardNodeAffinity approach, pods that have
+no interest in those special nodes need to know about all of the groups of special nodes.
+
+One final comment: in practice, it is often desirable to not
+only keep "regular" pods off of special nodes, but also to keep "special" pods off of
+regular nodes. An example in the dedicated nodes case is to not only keep regular
+users off of dedicated nodes, but also to keep dedicated users off of non-dedicated (shared)
+nodes. In this case, the "non-dedicated" nodes can be modeled as their own dedicated node group
+(for example, tainted as `dedicated=shared`), and pods that are not given access to any
+dedicated nodes ("regular" pods) would be given a toleration for `dedicated=shared`. (As mentioned earlier,
+we expect tolerations will be added by an admission controller.) In this case taints/tolerations
+are still better than hard affinity because with taints/tolerations each pod only needs one special "marking",
+versus in the hard affinity case where every time you add a dedicated node group (i.e. a new
+`dedicated=` value), you need to add a new hard affinity rule to all pods (including pending pods)
+except the ones allowed to use that new dedicated node group.
+
+## API
+
+```go
+// The node this Taint is attached to has the effect "effect" on
+// any pod that that does not tolerate the Taint.
+type Taint struct {
+	Key string  `json:"key" patchStrategy:"merge" patchMergeKey:"key"`
+	Value string  `json:"value,omitempty"`
+	Effect TaintEffect  `json:"effect"`
+}
+
+type TaintEffect string
+
+const (
+	// Do not allow new pods to schedule unless they tolerate the taint,
+	// but allow all pods submitted to Kubelet without going through the scheduler
+	// to start, and allow all already-running pods to continue running. 
+	// Enforced by the scheduler.
+	TaintEffectNoSchedule TaintEffect = "NoSchedule"
+	// Like TaintEffectNoSchedule, but the scheduler tries not to schedule
+	// new pods onto the node, rather than prohibiting new pods from scheduling
+	// onto the node. Enforced by the scheduler.
+	TaintEffectPreferNoSchedule TaintEffect = "PreferNoSchedule"
+	// Do not allow new pods to schedule unless they tolerate the taint,
+	// do not allow pods to start on Kubelet unless they tolerate the taint,
+	// but allow all already-running pods to continue running.
+	// Enforced by the scheduler and Kubelet.
+	TaintEffectNoScheduleNoAdmit TaintEffect = "NoScheduleNoAdmit"
+	// Do not allow new pods to schedule unless they tolerate the taint,
+	// do not allow pods to start on Kubelet unless they tolerate the taint,
+	// and evict any already-running pods that do not tolerate the taint.
+	// Enforced by the scheduler and Kubelet.
+	TaintEffectNoScheduleNoAdmitNoExecute = "NoScheduleNoAdmitNoExecute"
```

Comment:
I think some people might mistakenly think that the "No" binds only to "Schedule" rather than to "ScheduleAdmitExecute"